structure/parser/object_parser.go

```diff
@@ -202,28 +202,30 @@ func (o *Object) StringArray(reference string) *[]string {
 }
 
 func (o *Object) Time(reference string, layout string) *time.Time {
-	var timeValue time.Time
-	var err error
-
 	rawValue, ok := o.raw(reference)
 	if !ok {
 		return nil
 	}
 
-	switch rawValue.(type) {
-	case time.Time:
-		timeValue = rawValue.(time.Time)
-	default:
-		stringValue, ok := rawValue.(string)
-		if !ok {
-			o.base.WithReference(reference).ReportError(ErrorTypeNotTime(rawValue))
-			return nil
-		}
+	timeValue, timeValueOk := rawValue.(time.Time)
```

Comment:
should the object and array parser just use the same time parsing logic?

Reply:
Yea, probably. Perhaps with a common parsing implementation between the two, but that is a wider ranging change.